src/analysis_utils.py

```diff
@@ -0,0 +1,316 @@
+"""
+This module contains all the functions responsible for performing the analysis in this
+project.
```

Comment:
Consider being tiny more explicit on what you mean by analysis. Maybe include the names of the analysis in one sentence? Just a thought :)